lib/Traits/CswCatalogGroupTraits.ts

```diff
@@ -0,0 +1,166 @@
+  })
+  includeWms: boolean = true;
+
+  @primitiveTrait({
+    type: "boolean",
+    name: "Include KML",
+    description:
+      "True to allow KML resources to be added to the catalog; otherwise, false."
+  })
+  includeKml: boolean = true;
+
+  @primitiveTrait({
+    type: "boolean",
+    name: "Include CSV",
+    description:
+      "True to allow CSV resources to be added to the catalog; otherwise, false."
+  })
+  includeCsv: boolean = true;
+
+  @primitiveTrait({
+    type: "boolean",
+    name: "Include ESRI Map Server",
+    description:
+      "True to allow ESRI Map resources to be added to the catalog; otherwise, false."
+  })
+  includeEsriMapServer: boolean = true;
+
+  @primitiveTrait({
+    type: "boolean",
+    name: "Include GeoJSON",
+    description:
+      "True to allow GeoJSON resources to be added to the catalog; otherwise, false"
+  })
+  includeGeoJson: boolean = true;
+
+  @primitiveTrait({
+    type: "string",
+    name: "WMS Resource Format",
+    description:
+      "Gets or sets a regular expression that, when it matches the protocol attribute of a URI element of a record, indicates that the URI is a WMS resource."
+  })
+  wmsResourceFormat = "\\bwms\\b";
+
+  @primitiveTrait({
+    type: "string",
+    name: "KML Resource Format",
+    description:
+      "Gets or sets a regular expression that, when it matches the protocol attribute of a URI element of a record, indicates that the resource is a KML resource."
+  })
+  kmlResourceFormat = "\\bkml\\b";
+
+  @primitiveTrait({
+    type: "string",
+    name: "CSV Resource Format",
+    description:
+      "Gets or sets a regular expression that, when it matches the protocol attribute of a URI element of a record, indicates that the URI is a CSV resource."
+  })
+  csvResourceFormat = "\\bcsv-geo-";
+
+  @primitiveTrait({
+    type: "string",
+    name: "ESRI MapServer Resource Format",
+    description:
+      "Gets or sets a regular expression that, when it matches the protocol attribute of a URI element of a record, indicates that the URI is a Esri MapServer resource."
+  })
+  esriMapServerResourceFormat = "\\besri rest\\b";
+
+  @primitiveTrait({
+    type: "string",
+    name: "WMS Resource Format",
+    description:
+      "Gets or sets a regular expression that, when it matches the protocol attribute of a URI element of a record, indicates that the URI is a GeoJSON resource."
+  })
+  geoJsonResourceFormat = "\\bgeojson\\b";
+
+  @primitiveTrait({
+    type: "string",
+    name: "GetRecords Template",
+    description:
+      "Gets or sets the template XML string to POST to the CSW server to query for catalog items.  If this property is undefined,`lib/Models/CswGetRecordsTemplate.xml` is used.  The XML string should have a `{startPosition}` placeholder to be replaced with the next start position in order to allow incremental paging of results.."
+  })
+  getRecordsTemplate?: string;
+
+  // @primitiveTrait({
```

Comment:
Probably just delete this comment if it isn't needed?

Reply:
done